server/types/varchar.go

```diff
@@ -15,317 +15,93 @@
+	NotNull:       false,
+	BaseTypeOID:   0,
+	TypMod:        -1,
+	NDims:         0,
+	TypCollation:  100,
+	DefaulBin:     "",
+	Default:       "",
+	Acl:           nil,
+	Checks:        nil,
+	AttTypMod:     -1,
+	internalName:  "character varying",
+}
+
+// NewVarCharType takes maxChars representing the maximum number of characters that the type may hold
+func NewVarCharType(maxChars int32) (DoltgresType, error) {
+	var err error
+	newType := VarChar
+	newType.AttTypMod, err = GetTypModFromMaxChars("varchar", maxChars)
 	if err != nil {
-		return "", err
-	}
-	if b.IsUnbounded() {
-		return converted.(string), nil
-	}
-	str, _ := truncateString(converted.(string), b.MaxChars)
-	return str, nil
-}
-
-// IsPreferredType implements the DoltgresType interface.
-func (b VarCharType) IsPreferredType() bool {
-	return false
-}
-
-// IsUnbounded implements the DoltgresType interface.
-func (b VarCharType) IsUnbounded() bool {
-	return b.MaxChars == stringUnbounded
-}
-
-// Length implements the sql.StringType interface.
-func (b VarCharType) Length() int64 {
-	return int64(b.MaxChars)
-}
-
-// MaxByteLength implements the sql.StringType interface.
-func (b VarCharType) MaxByteLength() int64 {
-	return b.Length() * 4 // TODO
-}
-
-// MaxCharacterLength implements the sql.StringType interface.
-func (b VarCharType) MaxCharacterLength() int64 {
-	return b.Length()
-}
-
-// MaxSerializedWidth implements the DoltgresType interface.
-func (b VarCharType) MaxSerializedWidth() types.ExtendedTypeSerializedWidth {
-	if b.MaxChars != stringUnbounded && b.MaxChars <= stringInline {
-		return types.ExtendedTypeSerializedWidth_64K
-	} else {
-		return types.ExtendedTypeSerializedWidth_Unbounded
-	}
-}
-
-// MaxTextResponseByteLength implements the DoltgresType interface.
-func (b VarCharType) MaxTextResponseByteLength(ctx *sql.Context) uint32 {
-	if b.MaxChars == stringUnbounded {
-		return math.MaxUint32
-	} else {
-		return b.MaxChars * 4
-	}
-}
-
-// OID implements the DoltgresType interface.
-func (b VarCharType) OID() uint32 {
-	return uint32(oid.T_varchar)
-}
-
-// Promote implements the DoltgresType interface.
-func (b VarCharType) Promote() sql.Type {
-	return VarChar
-}
-
-// SerializedCompare implements the DoltgresType interface.
-func (b VarCharType) SerializedCompare(v1 []byte, v2 []byte) (int, error) {
-	if len(v1) == 0 && len(v2) == 0 {
-		return 0, nil
-	} else if len(v1) > 0 && len(v2) == 0 {
-		return 1, nil
-	} else if len(v1) == 0 && len(v2) > 0 {
-		return -1, nil
+		return DoltgresType{}, err
 	}
-	return serializedStringCompare(v1, v2), nil
+	return newType, nil
 }
 
-// SQL implements the DoltgresType interface.
-func (b VarCharType) SQL(ctx *sql.Context, dest []byte, v any) (sqltypes.Value, error) {
-	if v == nil {
-		return sqltypes.NULL, nil
-	}
-	value, err := b.IoOutput(ctx, v)
+// MustCreateNewVarCharType panics if used with out-of-bound value.
+func MustCreateNewVarCharType(maxChars int32) DoltgresType {
+	var err error
```

Comment:
This should call `NewVarCharType` and panic if an error is returned. This way, if we change the logic in `NewVarCharType`, we don't have to also remember to change it here too.